cloudevents/extensions/loglevel.md

```diff
@@ -0,0 +1,167 @@
+# Log Level Extension
+
+## Abstract
+This extension defines attributes that MAY be included within a CloudEvent
+to describe the "level" or "importance" of an event in relation to other events
+produced by a unique event source.
+
+Often systems produce events in form of logs, and these types of events usually share
+a common concept of "log-level". This extension aims to provide a standard way for 
+describing this property in a language agnostic form. 
+
+Sharing a common way to describe importance of events allows for better monitoring 
+systems, tooling and general log consumption.
+
+In addition to defining the extension attributes, this document defines 
+the concept of [Log Level Bindings](#log-level-bindings) - a standard way to map 
+well-known log level names from different languages and frameworks 
+onto the `loglevelname` and `loglevelnum` values defined in this spec. 
+
+## Attributes
+
+All attributes of this extension are OPTIONAL. however, when using this 
+custom extension you MUST use at-least one of these OPTIONAL attributes. 
+
+### loglevelname 
+
+- Type: `String`
+- Description: Human readable representation of the log level
+- Constraints
+  - OPTIONAL
+  - if present, MUST be a non-empty string
+  - SHOULD be lowercase
+  - RECOMMENDED values include `emergency`, `alert`, `critical`, `error`, 
+    `warning`, `notice`, `info`, `debug`, and `verbose`, but  others MAY be used.
+  - SHOULD have a one-to-one relationship with a distinct `loglevelnum` in the scope of
```

Comment:
It's not clear to me what this means - in many cases I'd expect a given source to only ever use one of them

Reply:
I can give you an example how this MAY happen.
I as an implementer of the binding for the python logging framework to this extension. Decide to go against the recommendations of the spec and place the python levelnames directly into the `loglevelname`.
The problem is in python you have two levelnames for python levelnum `30`  - `WARN` and `WARNING`.
If we decide that this constraint is a MUST this implementation MAY not be conformant. 
But I want to keep it in here so implementers will watch out for edge cases similar to this.

Reply:
Okay, so I think it needs a bit of rewording to avoid that - but without saying there has to be a 1:1 relationship. It should be valid for a source to *never* use loglevelname, or alternatively *never* use loglevelnumber, which at the moment would be a violation as there wouldn't be a 1:1 relationship.

Reply:
Ok I think I worded this in a confusing manner - the 1:1 relationship is with loglevelname and loglevelnum **values** and not the attributes themself. 
Meaning you cannot have
`info=1`
`error=1`
Or 
`info=2`
`info=3`

I think I can define the relation even better. 

The relation between all the values of `loglevelname` and all the values of `loglevelnum` in the given source SHOULD a one-to-one and onto 
ם

Reply:
But I'm saying it *doesn't* need to be. If I *only ever* specify a `loglevelname`, there's no 1:1 relationship (otherwise you'd be able to tell me which numeric value corresponds to each name, and you can't do that if I've never specified any numbers).

I believe I understand the constraint you're looking for - the same source should never use multiple numbers for the same name, or multiple names for the same number. I just don't think that's the same as saying there's a 1:1 relationship, when both are optional.

Reply:
@jskeet you are correct, I changed the wording to include this constraint only if both of the attributes are used